### PR TITLE
`TIMESTAMP <=> DATE` fail at bind time

### PR DESCRIPTION
**Pull request by romainfrancois**
related to #52

Now, the exception is thrown at bind time, i.e. during `rel_project()` thanks to the `bind` argument of `ScalarFunction()`. 

``` r
library(duckdbrfuns)

con <- duckdbrfuns:::duckdb_con()

time <- as.POSIXct(Sys.Date(), tz = "UTC")
date <- Sys.Date()

in_df <- tibble::tibble(x1 = time, x2 = date)
in_rel <- duckdb:::rel_from_df(con, in_df)

exprs <- list(
  duckdb:::expr_function(
    "r_base::==",
    list(
      duckdb:::expr_reference("x1"),
      duckdb:::expr_reference("x2")
    )
  )
)

out_rel <- duckdb:::rel_project(in_rel, exprs)
#> Error: {"exception_type":"Invalid Input","exception_message":"Comparing times and dates is not supported : TIMESTAMP <=> DATE"}
```

<sup>Created on 2024-04-19 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

File: duckdb-rfuns-r/R/relop.R [hunk 1]
```diff
@@ -56,25 +56,39 @@ spaceship_r <- function(x, y, ops = c("==", "!=", "<", "<=", ">", ">="), keep.da
...
-  spaceship <- as_tibble(
-    map(duckdb$rel_to_altrep(proj), \(col) col[])
   )
+
+  proj <- withCallingHandlers(
+    duckdb$rel_project(rel1, exprs),
+    error = function(err) {
+      cli_abort("binding error", call = error_call, parent = err)
+    }
+  )
```
Comment (romainfrancois) [inline duckdb-rfuns-r/R/relop.R:81]: We have to fo that sort of dance so that we can track where the error took place.

Comment (romainfrancois) [reply duckdb-rfuns-r/R/relop.R:81]: Perhaps it would be interesting for upstream `rel_project` to throw more sophisticated errors.

**Review by krlmlr — APPROVED** [review]
Great, thanks!

**Comment by krlmlr** [comment]
856 failing tests 🙀 🙀 🙀

**Comment by romainfrancois** [comment]
> 856 failing tests 🙀 🙀 🙀

I believe this is just a `duckdb-r` version mismatch, see #74

**Pull request merged**